docs/design/jet/023-jobupload.md

```diff
@@ -0,0 +1,163 @@
+| snapshotName  | String      | Argument passed when starting the job                     |
+| jobName       | String      | Argument passed when starting the job                     |
+| mainClass     | String      | Argument passed when starting the job                     |
+| jobParameters | List_String | Argument passed when starting the job                     |
+
+
+Upon reception of uploadJobMetaData message, the server performs validation. If any validation rule them fails, a **JetException** is thrown. If the message can be validated, the server stores a new entry in the JobUploadStore class
+
+**2. uploadJobMultipart message**
+
+The upload process continues with this message. It contains jar's bytes. This message contains these fields
+
+| Term              | Type      | Definition                                                                  |
+|-------------------|-----------|-----------------------------------------------------------------------------|
+| sessionId         | UUID      | Explained in the previous message                                           |
+| currentPartNumber | int       | Starts from 1 and shows the sequence number of the part. For example 1 of 5 |
+| totalPartNumber   | int       | The total number of parts of the sequence                                   |
+| partData          | byteArray | The **byte[]** containing jar data                                          |
+| partSize          | int       | Shows how many bytes of the partData byte[] is valid,                       |
+| sha256Hex         | String    | Hexadecimal SHA256 of the part                                              |
+
+**Why do we need an extra partSize field?** 
+
+For optimization, it is assumed that partData is allocated only once on the client side. So we need another field to indicate the number of bytes to be read from this buffer 
+
+Upon reception of uploadJobMultipart, various checks are performed on the message and current session. If any of them fail, a **JetException** is thrown. If the message can be validated, it is processed.
+Some checks are
+- Validate partData.length is positive
+- Validate partSize field is positive
+- Validate partSize == partData.length
+- Validate currentPart is not >= receivedCurrentPart - A message from the past
+- Validate currentPart + 1 is not different from receivedCurrentPart - A message from the future. This also means that a duplicate message will be rejected and upload operation will fail.
+- Validate totalPart != 0 && totalPart != receivedTotalPart
+- Validate checksum
+Upon reception of the first message a new temporary file is created. For every message the partData byte[] is appended to this file. The length of partData field is specified by the partSize field
+  When all the parts are complete, a new job is started using HazelcastBootstrap.executeJar() call. This call executes the jar within the same JVM as the member. 
```

Comment:
For the Hazelcast Platform Operator, we would like to have the possibility to execute a job from the jar that is already stored on the member. The reason is that the Operator already has the [ability to upload the files to the members](https://docs.hazelcast.com/operator/5.5/user-code-deployment), and it would be more nit to reuse the same functionality.
The scenario would look like this:
1. Operator uses its functionality to download the JAR from the S3 bucket and uploads it to the member.
2. `uploadJobMetaData` message is called to update the `JobUploadStore`
3. Having a new message that can start a new job from the existing JAR. Or an alternative could be that the `uploadJobMetaData` starts a new job if the JAR is already present. Both scenarios would work well for us

WDYT?